demo/monitoring-all-process-instances/js/frequency-execution-data.js

```diff
@@ -1,179 +1,185 @@
-let shapeFrequencyLegendStyles = new Map();
-let edgeFrequencyLegendStyles = new Map();
-
-function updateFrequencyTitleLegend(legendType, titles) {
-    let ticks = document.getElementById(`${legendType}-guide-y`).children;
-    ticks[0].firstElementChild.innerText = 0;
-    for (let i = 1; i < ticks.length; i++) {
-        ticks[i].firstElementChild.innerText = titles[i - 1];
-    }
-}
+let shapeFreqLegend;
+let edgeFreqLegend;
+let edgePathFreqLegend;
 
-function getLegendTitles(legendsStyles) {
-    return Array.from(legendsStyles.keys());
-}
 
-function updateFrequencyLegend(legendType, legendsStyles) {
-    const titles = getLegendTitles(legendsStyles);
-    document.getElementById(`${legendType}-250`).style.backgroundColor = legendsStyles.get(titles[0]);
-    document.getElementById(`${legendType}-200`).style.backgroundColor = legendsStyles.get(titles[1]);
-    document.getElementById(`${legendType}-150`).style.backgroundColor = legendsStyles.get(titles[2]);
-    document.getElementById(`${legendType}-100`).style.backgroundColor = legendsStyles.get(titles[3]);
-    document.getElementById(`${legendType}-50`).style.backgroundColor = legendsStyles.get(titles[4]);
-    updateFrequencyTitleLegend(legendType, titles);
+function buildLegendColors(styles) {
+    const legendsColors = new Array();
+    styles.forEach((values) => legendsColors.push(values.style.fill.color))
+    return legendsColors;
 }
 
 function updateFrequencyLegends() {
-    updateFrequencyLegend("shape-legend", shapeFrequencyLegendStyles);
-    updateFrequencyLegend("edge-legend", edgeFrequencyLegendStyles);
-    updateFrequencyTitleLegend("edge-path-legend", getLegendTitles(edgeFrequencyLegendStyles));
+    shapeFreqLegend.update();
+    edgeFreqLegend.update();
+    edgePathFreqLegend.update();
 }
 
 function withStrokeColorAsFillColor(overlayStyle) {
     return {...overlayStyle, stroke: {color: overlayStyle.fill.color}};
 }
 
-function getFrequencyOverlayStyles(position, color) {
-    return new Map([
-        ['random', {
+function buildFrequencyOverlayStyles(titles, position, color) {
+    return sortMap(new Map([
+        [titles[5], {
```

Comment:
ℹ️ usage of the array: it is hard to understand which colors are set for which overlay style as there is no more reference to the name of the style in this function

Reply:
I agree, but for the PR https://github.com/process-analytics/bpmn-visualization-examples/pull/230, it make easier to avoid duplication.

Reply:
ok, I will check this while reviewing PR #230